tests/system/Validation/ValidationTest.php

```diff
@@ -270,6 +270,16 @@ public function testRunDoesTheBasics(): void
         $this->assertSame([], $this->validation->getValidated());
     }
 
+    public function testRunWithNumericFieldName(): void
+    {
+        // The array key will be int.
+        $data = ['123' => 'notanumber'];
```

Comment:
Unless this has changed recently, PHP will cast string j refer array keys to integers, so this is functionally identical to `[123 => 'notanumber']`

Reply:
Yes, PHP does not allow numeric string keys in arrays.

Reply:
I used `'123'` in the test code because most field names are strings, so I guess many devs would write like that code.
But after all PHP will have an integer key.